src/java.base/share/classes/java/net/Socket.java

```diff
@@ -532,7 +532,7 @@ void createImpl(boolean stream) throws SocketException {
             impl.create(stream);
             created = true;
         } catch (IOException e) {
-            throw new SocketException(e.getMessage());
+            throw new SocketException(e.getMessage(), e);
```

Comment:
For this one, you could `catch (SocketException e) { throw e; }` as the exception from the impl will be a SocketException anyway. That would help avoid the confusing cause.

Reply:
Hi Alan, createImpl  thros SocketException anyway, so can't we just remove the catch completely in this case ?
Or are there other IOExceptions?

Reply:
Our SocketImpls mostly throw SocketException so inserting the catch of SocketException will avoid the translation for most cases.

Reply:
Replacing the IOException does not work, this even leads to a compile error " unreported exception IOException" ; so I guess the catch (SocketException e)  needs to be added to the IOException ; not so nice.

Reply:
I didn't mean to replace it, I meant to insert of catch of SocketException before the IOException. That will avoid throwing SocketException with cause SocketException. If you look at the equivalent in ServerSocket then you'll see what I mean.